### PR TITLE
SC-2661: Support blackfire

### PR DESCRIPTION
**Pull request by zyuzka**
Epic: https://spryker.atlassian.net/browse/SC-2661

##### Change log:
- Added Blackfire support.
- Added NewRelic support.

### CODE REVIEW

File: generator/index.php [hunk 1]
```diff
@@ -450,3 +450,49 @@ function isHost(string $knownHost): bool
...
+    if (!isBlackFireEnabled($blackfireConfig)) {
+        return [];
+    }
+
+    validateBlackfireConfig($blackfireConfig);
+
+    return $blackfireConfig;
+}
+
+function isBlackFireEnabled(array $blackfireConfig): bool
```
Comment (alexanderM91) [inline generator/index.php]: Please remove one line method

File: generator/index.php [hunk 2]
```diff
@@ -450,3 +450,49 @@ function isHost(string $knownHost): bool
...
+        return [];
+    }
+
+    $blackfireConfig = $projectData['docker']['blackfire'];
+
+    if (!isBlackFireEnabled($blackfireConfig)) {
+        return [];
+    }
+
+    validateBlackfireConfig($blackfireConfig);
```
Comment (alexanderM91) [inline generator/index.php]: maybe

if () 
throw

File: images/base_app/Dockerfile [hunk 3]
```diff
@@ -7,6 +7,8 @@ RUN mkdir -p /cache && chmod 0777 /cache
 
 # Intermediate container - base for app, with composer deps
 FROM ${SPRYKER_PLATFORM_IMAGE} AS spryker_app_dependencies
+# todo: env
```
Comment (alexanderM91) [inline images/base_app/Dockerfile]: todo?

File: generator/src/templates/docker-compose.yml.twig [hunk 4]
```diff
@@ -5,6 +5,27 @@ version: "3.5"
 } %}
 
 services:
+{% if _blackfire['enabled'] %}
```
Comment (alexanderM91) [inline generator/src/templates/docker-compose.yml.twig]: IMO it better to move registration inside service section and create a separate blackfire.service.twig

File: images/base_app/Dockerfile (hunk 3 above)
Comment (sprymiker) [reply images/base_app/Dockerfile]: It should be turnable based on deploy.yml

File: images/base_dev/Dockerfile [hunk 5]
```diff
@@ -14,6 +14,10 @@ ENV DEVELOPMENT_CONSOLE_COMMANDS=1
 
 ENV PATH=/data/vendor/bin:$PATH
 
+ARG BLACKFIRE_ENABLED
+ENV BLACKFIRE_ENABLED=${BLACKFIRE_ENABLED}
+RUN bash -c '[ ${BLACKFIRE_ENABLED} -eq 0 ] && mv /usr/local/etc/php/conf.d/blackfire.ini.dist /usr/local/etc/php/conf.d/blackfire.ini || true'
```
Comment (sprymiker) [inline images/base_dev/Dockerfile]: [ "${BLACKFIRE_ENABLED}" == "1" ]

File: bin/images/baked.sh [hunk 6]
```diff
@@ -38,6 +38,7 @@ function doBaseImage()
         --build-arg SPRYKER_LOG_DIRECTORY=${logDirectory} \
         --build-arg SPRYKER_COMPOSER_MODE=${SPRYKER_COMPOSER_MODE} \
         --build-arg SPRYKER_COMPOSER_AUTOLOAD=${SPRYKER_COMPOSER_AUTOLOAD} \
+        --build-arg BLACKFIRE_ENABLED=${BLACKFIRE_ENABLED:-1} \
```
Comment (sprymiker) [inline bin/images/baked.sh]: ${BLACKFIRE_ENABLED:-""}

File: bin/images/mount.sh [hunk 7]
```diff
@@ -61,6 +61,7 @@ function buildBaseImages()
         --build-arg SPRYKER_LOG_DIRECTORY=${logDirectory} \
         --build-arg SPRYKER_COMPOSER_MODE=${SPRYKER_COMPOSER_MODE} \
         --build-arg SPRYKER_COMPOSER_AUTOLOAD=${SPRYKER_COMPOSER_AUTOLOAD} \
+        --build-arg BLACKFIRE_ENABLED=${BLACKFIRE_ENABLED:-1} \
```
Comment (sprymiker) [inline bin/images/mount.sh]: ${BLACKFIRE_ENABLED:-""}

File: generator/index.php (hunk 2 above)
Comment (sprymiker) [reply generator/index.php]: Please rename to assert

File: generator/src/templates/deploy.bash.twig [hunk 8]
```diff
@@ -152,6 +152,28 @@ DOCKER_EXTRA_OPTIONS+=" -v ${SPRYKER_DOCKER_PREFIX}_${SPRYKER_DOCKER_TAG}_data_s
 {% endif %}
 {% endif %}
 
+{% if _blackfire['enabled'] %}
+export BLACKFIRE_ENABLED={{ _blackfire['enabled'] ? 0 : 1 }}
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig]: readonly instead of export

File: generator/src/templates/deploy.bash.twig [hunk 9]
```diff
@@ -512,6 +534,9 @@ function doCli()
     local binFile=$1
     local binFileArgument=${@:2}
 
+{% if _blackfire['enabled'] %}
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig]: if [ -n "${BLACKFIRE_ENABLED}" ]; then....

File: generator/src/templates/deploy.bash.twig (hunk 8 above)
Comment (sprymiker) [reply generator/src/templates/deploy.bash.twig]: ="{{ _blackfire['enabled'] ? 1 : "" }}"

File: generator/src/templates/deploy.bash.twig [hunk 10]
```diff
@@ -152,6 +152,28 @@ DOCKER_EXTRA_OPTIONS+=" -v ${SPRYKER_DOCKER_PREFIX}_${SPRYKER_DOCKER_TAG}_data_s
 {% endif %}
 {% endif %}
 
+{% if _blackfire['enabled'] %}
+export BLACKFIRE_ENABLED={{ _blackfire['enabled'] ? 0 : 1 }}
+export BLACKFIRE_SERVER_ID=${BLACKFIRE_SERVER_ID:-{{ _blackfire['server-id'] }}}
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig]: "...

File: generator/src/templates/docker-compose.yml.twig (hunk 4 above)
Comment (sprymiker) [reply generator/src/templates/docker-compose.yml.twig]: Let's leave it for now until refactoring.

File: images/base_app/Dockerfile [hunk 11]
```diff
@@ -8,6 +8,10 @@ RUN mkdir -p /cache && chmod 0777 /cache
 # Intermediate container - base for app, with composer deps
 FROM ${SPRYKER_PLATFORM_IMAGE} AS spryker_app_dependencies
 
+ARG BLACKFIRE_ENABLED
+ENV BLACKFIRE_ENABLED=${BLACKFIRE_ENABLED}
```
Comment (sprymiker) [inline images/base_app/Dockerfile]: Check if ENV is necessary

**Pull request merged**